tracer/src/Datadog.Trace/Debugger/Expressions/ProbeProcessor.cs

```diff
@@ -393,7 +393,7 @@ private void CheckSpanDecoration(DebuggerSnapshotCreator snapshotCreator, ref bo
                         {
                             Tracer.Instance.ScopeManager.Active.Root.Span.SetTag(evaluationErrorTag, string.Join(";", decoration.Errors));
                         }
-                        else if (Tracer.Instance.ScopeManager.Active.Span.GetTag(evaluationErrorTag) != null)
+                        else if (Tracer.Instance.ScopeManager.Active.Root.Span.GetTag(evaluationErrorTag) != null)
```

Comment:
Somewhat unrelated, but

- You probably shouldn't use `Tracer.Instance.ScopeManager.Active` - you should use `Tracer.Instance.InternalActiveScope` _or_ if you want the "distributed" scope details (in version-mismatch scenarios with V2), use `Tracer.Instance.ActiveScope`.
- I would cache the access locally in this method rather than repeatedly accessing it
- This can return `null`, so you should probably be checking for that?

Reply:
Thanks for that!

Reply:
I'm reverting this fix and I'll addressing it and all the other things you have mentioned in a different PR (soon).

Reply:
https://github.com/DataDog/dd-trace-dotnet/pull/6242